### DISCUSSION TITLE
Can/Does 'juicefs' work like HDFS?

### DISCUSSION THREAD

**Discussion by seigel** [juicedata/juicefs#51] [Q&A] [answered]
My question is about the idea of 'connecting' multiple clouds behind a single mount point like `~/jfs` such that stored blocks would be distributed to multiple clouds according to some algorithm.

1 block to S3, next block to GCS or something.  Allow the option to have copies so if a cloud is unavailable, things still work. Distribute by economical model (price of storage), etc...

Anyway, I was curious.  My goal interest is encrypted blocks across multiple clouds for security, reliability and avoid storage limitations.

Great Work everyone!

**Comment by davies** [comment, marked as answer]
Currently, JuiceFS does not do this, but it's easy to implement this. You can have a virtual object store which distribute object to others based on hash or suffix.

This is an interesting idea. Someone has tried that, but the performance could be really bad because the limited bandwidth across cloud providers.
